spec/struct.dd

```diff
@@ -473,31 +473,58 @@ $(H2 $(LEGACY_LNAME2 StructLiteral, struct-literal, Struct Literals))
         int foo(S s) { return s.x; }
 
         foo( S(1, 2) ); // set field x to 1, field y to 2
+        foo( S(y: 2, x: 1) ); // same as above
         ---
         )
 
         $(P Struct literals are syntactically like function calls.
-        If a struct has a member function named $(CODE opCall), then
+        If a struct has a $(RELATIVE_LINK2 struct-constructor, Constructor)
+        or a member function named `opCall`, then
         struct literals for that struct are not possible. See also
         $(DDSUBLINK spec/operatoroverloading, FunctionCall, opCall operator overloading)
         for the issue workaround.)
         $(P
-        It is an error if there are more arguments than fields of
-        the struct.
-        If there are fewer arguments than fields, the remaining
-        fields are initialized with their respective default
-        initializers.)
+            If the first argument has no name, it will be assigned to the struct field that is defined first lexically.
+        )
+        $(P
+            A named argument is assigned to the struct field with the same name.
+            It is an error if no such field exists.
+        )
+        $(P
+            Any other argument is assigned to the next lexically defined struct field relative to the preceding argument's struct field.
+            It is an error if no such field exists, i.e. when the preceding argument assigns to the last struct field.
+        )
+        $(P
+            It is also an error to assign a field more than once.
+        )
         $(P
-        If there is a union field in the struct, only the first
+            Any fields not assigned a value are initialized with their respective default initializers.
+        )
+        $(NOTE
+            These rules are consistent with function calls, see $(DDSUBLINK spec/function, argument-parameter-matching, Matching Arguments to Parameters).
+        )
+        $(P
+        If there is a union field in the struct, only one
         member of the union can be initialized inside a
         struct literal. This matches the behaviour for union literals.
         )
 
+        $(SPEC_RUNNABLE_EXAMPLE_FAIL
+        ---
+        struct S { int x = 1; int y = 2, int z = 3; }
+
+        S s0 = S(y: 5, 6, x: 4); // `6` is assigned to field `z`, which comes after `y`
+        S s1 = S(y: 5, z: 6);    // Field x is not assigned, set to default initializer `1`
+        S s2 = S(y: 5, x: 4, 5); // Error: field `y` is assigned twice
+        S s3 = S(z: 2, 3);       // Error: no field beyond `z`
```

Comment:
This line actually segfaults:
https://issues.dlang.org/show_bug.cgi?id=24281

This is why I don't like SPEC_RUNNABLE_EXAMPLE_FAIL, because it hides unwanted failures!